### ISSUE TITLE
Rendering a very coarse mesh with 10 vertices or fewer crashes GLMeshItem

### ISSUE THREAD

**Issue by gandreadis** [pyqtgraph/pyqtgraph#2604]
### Short description
When I try to render really coarse meshes (with <=10 vertices), I get an error message and the rendering crashes. The exact same code used to draw larger meshes works fine and renders as expected.

### Code to reproduce

```python
import pyqtgraph as pg
import numpy as np

vertices = [(0, 0, 0), (1, 1, 1), (0, 0, 1), (0, 1, 0)] 
faces = [(0, 1, 2), (0, 1, 3), (0, 2, 3), (1, 2, 3)]
gl.GLMeshItem(vertexes=vertices, faces=faces)
# ... standard code rendering this object in a scene
```

### Expected behavior
The render should proceed as it would for a more complex mesh.

### Real behavior
Rendering crashes and no mesh is displayed.

```
Error while drawing item <pyqtgraph.opengl.items.GLMeshItem.GLMeshItem object at 0x1684943a0>.
/Users/gandreadis/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pyqtgraph/opengl/GLViewWidget.py:259: RuntimeWarning: 
Traceback (most recent call last):
  File "[redacted]/gui/start_gui.py", line 22, in <module>
    sys.exit(app.exec())
  File "/Users/gandreadis/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pyqtgraph/opengl/GLViewWidget.py", line 74, in showEvent
    self._updateScreen(window.screen())
  File "/Users/gandreadis/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pyqtgraph/opengl/GLViewWidget.py", line 62, in _updateScreen
    self._updatePixelRatio()
  File "/Users/gandreadis/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pyqtgraph/opengl/GLViewWidget.py", line 69, in _updatePixelRatio
    self.resizeEvent(event)
  File "/Users/gandreadis/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pyqtgraph/opengl/GLViewWidget.py", line 238, in paintGL
    self.drawItemTree(useItemNames=useItemNames)
  File "/Users/gandreadis/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pyqtgraph/opengl/GLViewWidget.py", line 270, in drawItemTree
    self.drawItemTree(i, useItemNames=useItemNames)
  File "/Users/gandreadis/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pyqtgraph/opengl/GLViewWidget.py", line 259, in drawItemTree
    debug.printExc()
  --- exception caught here ---
  File "/Users/gandreadis/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pyqtgraph/opengl/GLViewWidget.py", line 256, in drawItemTree
    i.paint()
  File "/Users/gandreadis/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pyqtgraph/opengl/items/GLMeshItem.py", line 165, in paint
    self.parseMeshData()
  File "/Users/gandreadis/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pyqtgraph/opengl/items/GLMeshItem.py", line 141, in parseMeshData
    self.vertexes = md.vertexes(indexed='faces')
  File "/Users/gandreadis/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pyqtgraph/opengl/MeshData.py", line 124, in vertexes
    self._vertexesIndexedByFaces = self._vertexes[self.faces()]
IndexError: too many indices for array: array is 2-dimensional, but 4 were indexed
  debug.printExc()
```

### Tested environment(s)

 * PyQtGraph version: 0.13.1
 * Qt Python binding: PyQt5 5.15.6 Qt 5.15.2
 * Python version: 3.8.2
 * NumPy version: 1.21.6
 * Operating system: Mac OSX 12.6
 * Installation method: pip

### Additional context
--

**Comment by pijyoi** [comment]
Could you provide a minimum _runnable_ code that demonstrates the issue?

**Comment by pijyoi** [comment]
In the example found in https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/examples/GLMeshItem.py, numpy arrays are passed into `GLMeshItem` even though there are only 4 vertices and 4 faces.

Browsing through the source code, the code _does_ assume that the data passed in are numpy arrays and would not work at all for Python lists.

Are you passing in numpy arrays for your larger meshes but Python lists for your coarser meshes?

**Comment by gandreadis** [comment]
Hi @pijyoi, thanks for the quick response. I had tried numpy arrays at one point, but I see now that I had reversed that in one place. Bottom-line: with numpy arrays it now always works, with Python lists it only works with large meshes (maybe numpy indexing uses some internal threshold on length after which it just converts?). Problem solved for me, but may be good to explicitely say something about this in documentation? Because documentation for, e.g. [`GLMeshItem`](https://pyqtgraph.readthedocs.io/en/latest/api_reference/3dgraphics/glmeshitem.html) does not say anything about the format of arrays required.
